src/ILCompiler.ReadyToRun/src/Compiler/DependencyAnalysis/ReadyToRun/SignatureBuilder.cs

```diff
@@ -377,16 +391,21 @@ private void EmitArrayTypeSignature(ArrayType type, SignatureContext context)
                 flags |= (uint)ReadyToRunMethodSigFlags.READYTORUN_METHOD_SIG_Constrained;
             }
 
+            if (methodToken.IsNull)
+            {
+                methodToken = context.GetModuleTokenForMethod(method.GetMethodDefinition(), throwIfNotFound: false);
```

Comment:
Do we need to lookup both GetMethodDefinition and GetTypicalMethodDefinition here? It looks fishy.